go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -597,11 +599,13 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		// ensure we have reserved the IPs found in OVN
 		if len(podIfAddrs) == 0 {
 			needsNewAllocation = true
-		} else if err = bnc.lsManager.AllocateIPs(switchName, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
-			klog.Warningf("Unable to allocate IPs %s found on existing OVN port: %s, for pod %s on switch: %s"+
-				" error: %v", util.JoinIPNetIPs(podIfAddrs, " "), portName, podDesc, switchName, err)
+		} else if bnc.doesNetworkRequireIPAM() {
```

Comment:
So I think that `needsNewAllocation` really means `needsNewMacOrIPAllocation`. So I would rename the variable or put a comment somewhere because it can be confusing.

What happens when we handle a pod for IPAM less network that already has a MAC address annotated? Are we handling that correctly? Do we avoid allocating a different MAC? Do we need to check if that MAC is already in use?

Reply:
Would it make sense to generate a locally administrated MAC from a SHA256 of the pod's UUID ? 

Otherwise, we can always generate something, and retry if we see that address is used on the cluster wide logical switch. ... this would be more costly, as afaiu it would require more ovsdb reads .